Openmc 0.13.1 compatibility

Rademacher opened the pull request:
Made ONIX compatible with minor naming and formatting changes introduced with openmc 0.13.1 while retaining backward compatibility

Rademacher committed:
Compatibility with openmc >= 0.13.1
Adjusted ONIX to be compatible with newer openmc versions (specifically a changed return format of openmc.Material.get_nuclide_atom_densities()). Backwards compatibility is retained by checking openmc version.

Rademacher committed:
Remove outdated argument
Removed the obsolete argument 'mc_nuclide_densities' from cell._set_MC_tallies and couple._set_tallies_to_bucells.

Rademacher committed:
Adopt keff renaming
Adopted the renaming of k_combined to keff in openmc.statepoint with openmc 0.13.1 while retaining backwards compatibility

jung-benjamin commented on onix/couple/couple_openmc.py:
```diff
@@ -913,7 +916,10 @@ def _pass_nuclide_densities(self, cell_dict, bucell_dict):
                 # if nucl is one of the initial non-zero initial nuclide, pass the density
                 if nucl in init_nucl:
-                    onix_dens_dict[onix_nucl] = openmc_dens_dict[nucl][1]
+                    if ((int(openmc.__version__.split('.')[1]) == 13) and (int(openmc.__version__.split('.')[2]) >= 1)) or (int(openmc.__version__.split('.')[1]) > 13):
+                        onix_dens_dict[onix_nucl] = openmc_dens_dict[nucl]
+                    else:
+                        onix_dens_dict[onix_nucl] = openmc_dens_dict[nucl][1]
```
```suggestion
                    if isinstance(onix_dens_dict, dict):
                        onix_dens_dict[onix_nucl] = openmc_dens_dict[nucl]
                    else:
                        onix_dens_dict[onix_nucl] = openmc_dens_dict[nucl][1]
```

I think this makes the purpose clearer and is more readable. I believe it achieves the same functionality and may be more robust towards future changes of opened.

Rademacher replied:
While I agree with the sentiment of the explicit version check being quite unwieldy I think we should at least leave a comment referencing where this is coming from. 
Regarding the check itself, I think it needs to be slightly different to actually work as intended. And if our goal is to eliminate the unwieldy expression we might as well save the result in a variable and reuse it for the criticality, e.g. like this: 
```suggestion
                    # OpenMC changed the format of material.get_nuclide_atom_densities() from version 0.13.1 onwards
                    global old_openmc = False
                    if isinstance(openmc_dens_dict[nucl], tuple):
                        onix_dens_dict[onix_nucl] = openmc_dens_dict[nucl][1]
                        old_openmc = True
                    else:
                        onix_dens_dict[onix_nucl] = openmc_dens_dict[nucl]
```

jung-benjamin reviewed (approved):
The changes look solid, although I have suggested some improvements.

Rademacher commented on onix/couple/couple_openmc.py:
```diff
@@ -568,7 +568,10 @@ def _set_statepoint(self, path = os.getcwd()):
 
         statepoint = self.statepoint
-        kinf = statepoint.k_combined
+        if ((int(openmc.__version__.split('.')[1]) == 13) and (int(openmc.__version__.split('.')[2]) >= 1)) or (int(openmc.__version__.split('.')[1]) > 13):
+            kinf = statepoint.keff
+        else:
+            kinf = statepoint.k_combined
```
Its probably nicer to reuse the version check result from the density function:

```suggestion
        if old_openmc:
            kinf = statepoint.k_combined
        else:
            kinf = statepoint.keff
```

jung-benjamin replied:
I think your solution is good, although I dislike using global variables like that. An alternative option is to use a `try except` clause. I'm not sure if there are solid reasons to prefer one over the other.
```suggestion
        try:
            kinf = statepoint.keff
        except AttributeError:
            kinf = statepoint.k_combined
```
(I'm assuming it would throw an `AttributeError`. I haven't tested it.)
Feel free to commit whichever one you prefer.

Rademacher committed:
Switch to try/except
Switched to try/except constructions instead of the version number checks for easier readibility and potentially more robustness towards future changes.
